packages/fxa-auth-server/lib/oauth/db/mysql/index.js

```diff
@@ -181,7 +182,16 @@ const QUERY_ACCESS_TOKEN_INSERT =
 const QUERY_REFRESH_TOKEN_INSERT =
   'INSERT INTO refreshTokens (clientId, userId, email, scope, token, profileChangedAt) VALUES ' +
   '(?, ?, ?, ?, ?, ?)';
-const QUERY_ACCESS_TOKEN_FIND = 'SELECT * FROM tokens WHERE token=?';
+// Access token storage in redis annotates each token with client metadata,
+// so we do the same when reading from MySQL, for consistency.
+const QUERY_ACCESS_TOKEN_FIND =
+  'SELECT tokens.token, tokens.clientId, tokens.createdAt, ' +
+  '  tokens.userId, tokens.email, tokens.scope, ' +
+  '  tokens.createdAt, tokens.expiresAt, tokens.profileChangedAt, ' +
+  '  clients.name as clientName, clients.canGrant AS clientCanGrant, ' +
+  '  clients.publicClient ' +
+  'FROM tokens LEFT OUTER JOIN clients ON clients.id = tokens.clientId ' +
+  'WHERE tokens.token=?';
```

Comment:
This is a more expensive query, but we'll only do it for a small number of tokens, so IMHO the increase in internal consistency is worth it. I won't be upset if you push back though, and we could instead default the client-related fields to `null`.

Reply:
LGTM. clients is a tiny table